config/config.go

```diff
@@ -228,10 +229,40 @@ func (c *Config) LoadConfigFromContent(content []byte) (err error) {
 		return
 	}
 
+	err = c.parseEndpoint()
+	if err != nil {
+		return
+	}
+
 	c.InitHTTPClient()
 	return
 }
 
+func (c *Config) parseEndpoint() (err error) {
+	if c.Endpoint == "" {
+		return nil
+	}
+
+	parts := strings.Split(c.Endpoint, "://")
```

Comment:
Maybe we can use `url.Parse` instead of `strings.Split` here?

Reply:
OK